lib/clients/VaultClientCache.js

```diff
@@ -0,0 +1,133 @@
+const fs = require('fs');
+
+const VaultClient = require('vaultclient').Client;
+
+/**
+ * @class VaultClientCache
+ *
+ * @classdesc Helper class to manage a set of Vault clients
+ *
+ * Clients share the same configuration based on the profile they
+ * belong to. A profile is automatically created when it does not
+ * exist and the first configuration is set for a given profile name.
+ *
+ * When getClient() is called, a new or cached Vaultclient instance is
+ * returned, based on if the profile name, host and port provided
+ * match a cached instance. When VaultClient instances are created
+ * they are cached and kept indefinitely.
+ */
+class VaultClientCache {
+    constructor() {
+        this._profiles = {};
+        this._vaultClients = {};
+    }
+
+    _updateProfile(profileName, params) {
+        const profile = this._profiles[profileName] || {};
+        this._profiles[profileName] = profile;
+
+        Object.assign(profile, params);
+        return this;
+    }
+
+    /**
+     * Load Vault administrative credentials from a file, bind them to
+     * the profile given by profileName. Those credentials will be
+     * configured in further created clients under the same profile
+     * name.
+     *
+     * The function throws an error if the file is not readable or has
+     * invalid format.
+     *
+     * @param {string} profileName - name of profile to bind the
+     *   credentials to
+     * @param {string} filePath - path to credentials file. This file
+     *   must be a JSON file in the following format:
+     *   {"accessKey":"secretKey"}
+     * @return {VaultClientCache} this
+     */
+    loadAdminCredentialsFromFile(profileName, filePath) {
+        const adminCredsJSON = fs.readFileSync(filePath);
```

Comment:
This should happen in Config.json. Validate, parse and load into the config object.